### DISCUSSION TITLE
質問が削除されても回答したことの記録を残して欲しい

### DISCUSSION THREAD

**Discussion by shiracamus** [increments/qiita-discussions#230] [要望・アイデア・不具合]
質問に対して回答しても、質問を削除してしまう人がいて、せっかくの知見が共有されなくてとてもガッカリします。また、自分の質問回答一覧に記録が残らないのも悲しいです。
質問を削除するときに「削除せずにクローズして知見を共有しましょう」と促せないでしょうか。
せめて、誰のどんな質問（タイトル）に回答したかの記録を自分の「回答した質問」の一覧に残していただけないでしょうか。
質問へのリンクをクリックして Not Found になるのはしかたないですけど。

**Comment by tanaka-asami415** [comment]
@shiracamus 

ご意見ありがとうございます🙏

とても共感いたします…😢我々も課題に感じておりまして、改善したいと思っております！
具体的にどう変えるかなど検討が必要であり、少し先にはなってしまうと思いますが
少なくとも回答した記録は残せるようにしたいと考えています🙇

**Comment by radian-jp** [comment]
私も最近同じような事がありました。
回答のモチベーション自体も大幅に下がりますし、単なるコピペじゃなければ回答内容にも著作権は発生する筈なので、
他人が質問を削除して、回答内容も巻き込まれて削除されるのは非常に問題がある状態と考えます。
回答がついたら削除不可にするなり、何らかの対応をお願いしたいです。

**Reply by tanaka-asami415** [reply]
@radian-jp 
ご意見ありがとうございます🙏
こちら優先度を上げられないか検討いたします🙇

**Reply by radian-jp** [reply]
この件現在も改善されていないですが、何とかならないでしょうか？
学校の課題を質問に丸投げし、回答させて削除といった悪質な行為も頻繁に見受けられます。
回答した事が無駄になりますし、情報として蓄積されません。

**Reply by tanaka-asami415** [reply]
@radian-jp 
力及ばず申し訳ないです😢
質問の削除が選択された際に、アナウンスを出すようにしたのですが、
それでも消してしまう方がおられるようで、引き続き別の手法も検討中です。
すぐの解決が出来ず申し訳ありませんが、ご容赦いただけますと幸いです🙇

<details><summary>アナウンス画像</summary>

![qa](https://github.com/increments/qiita-discussions/assets/103916185/78839546-7ea0-457b-ae0b-f15050186f1d)
</details>

**Reply by radian-jp** [reply]
ユーザーの善意を前提とした対策は、ほぼ無意味だと思います。
削除したい人は絶対に削除します。
回答がついた時点で削除は出来なくするべきだと思います。

**Reply by tanaka-asami415** [reply]
@radian-jp 
ご意見ありがとうございます。
はい。ご提案いただいたような方法も合わせて検討いたします🙇

**Comment by tanaka-asami415** [comment]
@radian-jp
大変お待たせいたしました🙇 回答がついた質問は削除できないようにいたしました！
https://qiita.com/release-notes/2023-10-04

まだユーザーが退会してしまった場合は消えてしまう状態なのですが、こちらも追って対応を予定しております。

@shiracamus
回答がついた質問は削除できないようにいたしました！
記録を一覧に残す変更については、影響範囲が大きいため別途対応を検討しています🙇
